org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/JavaSemanticTokensProcessor.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Broadcom, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Broadcom, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.lsp4e.jdt;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jdt.ui.text.java.ISemanticTokensProvider;
+import org.eclipse.lsp4e.operations.semanticTokens.SemanticTokensDataStreamProcessor;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.SemanticTokensLegend;
+
+class JavaSemanticTokensProcessor {
```

Comment:
This class has a lot of common code with `org.eclipse.lsp4e.operations.semanticTokens.SemanticTokensDataStreamProcessor`, could we have a common implementation?